lib/new_relic/util/http.ex

```diff
@@ -42,8 +42,10 @@ defmodule NewRelic.Util.HTTP do
   https://erlef.github.io/security-wg/secure_coding_and_deployment_hardening/ssl
   """
   def http_options(opts \\ []) do
+    env_opts = Application.get_env(:new_relic_agent, :httpc_request_options, [])
+
     [
-      connect_timeout: 1000,
+      connect_timeout: 10_000,
```

Comment:
We could leave the existing 1000ms in place and let users raise this value by overriding it in the `env_opts`, correct?

If so let's revert back to 1000ms, I'm inclined to favor a minimal change

Reply:
Currently we have confirmation that users on both Heroku and EC2 are unable to use this plugin with the default 1000ms timeout. I'd hypothesis that other platforms would also then be seeing this. Having a sensible default which is known to work with NewRelic's very slow servers seems like a far better user experience.

Reply:
If we want to solve the problem of users needing to override the global httpc options then this PR, after backing out the time adjustment, looks good. Let's keep it focused on one thing.

If we want to solve the issue with contacting the collector I think we should solve that specifically in `NewRelic.Harvest.Collector.Protocol` instead of increasing the connect timeout of every request made with `NewRelic.Util.HTTP` globally. I think that should happen in another PR. 

My thinking is that PR would need to do a few things:
 * `NewRelic.Util.HTTP.post` needs an optional connect_timeout parameter that overrides the default
 * `NewRelic.Harvest.Collector.Protocol` can gain a module variable, something like `@collector_connect_timeout` set to `20_000` ([20s is the value the Go agent uses](https://github.com/newrelic/go-agent/blob/41d47c0d8dadd1f998ea46dec7a77183abfa042f/v3/newrelic/limits.go#L16) so we can follow in their footsteps)
 * `NewRelic.Harvest.Collector.Protocol` should pass that value in to `NewRelic.Util.HTTP.post` to override the default so that requests to the collector are guaranteed their full 20 second window to connect regardless of how httpc is globally configured.
 * All in all I think it'd be pretty quick, just a little plumbing.
 
 Both of these seem like great improvements!

Reply:
By the way, one thing I worry about with respect to increasing the timeout for the collector is a 20 second wait exceeds the default timeout of a GenServer.call (5 seconds) which could lead to some unexpected crashes of these genservers which I'm not sure we have graceful error handling for right now.

We might want to be more conservative in Collector.Protocol and only increase the timeout for `preconnect/0` and `connect/1` specifically, instead of setting it in `issue_call/2`. We can discuss that separately though.

Reply:
I've removed this change from the PR. Feel free to either merge it or close it. 

I also don't intend to make the changes you want for the timeout so feel free to pick that up yourself. Though note that the Harvester `send_harvest` GenServer call has a custom timeout of 15 seconds FWIW.

Reply:
Nice find on send_harvest. Sounds like it'll take a little tracing through the code to make sure we can find a good number to start with (or adjust some timeouts elsewhere)

> I also don't intend to make the changes you want for the timeout so feel free to pick that up yourself. 

Yeah, not a lot of time on my end either, maybe someone will be able to chase it in the future!